src/main/scala/com/microsoft/hyperspace/actions/CreateAction.scala

```diff
@@ -72,4 +75,40 @@ class CreateAction(
   // TODO: The following should be protected, but RefreshAction is calling CreateAction.op().
   //   This needs to be refactored to mark this as protected.
   final override def op(): Unit = write(spark, df, indexConfig)
+
+  final override protected def event(appInfo: AppInfo, message: String): HyperspaceEvent = {
+    // logEntry may or may not exist the first time an index is being created.
```

Comment:
Sorry I didn't phrase it right. I am just referring to the `logEntry` variable in this class.

So vacuum scenario won't occur because `val logEntry` is freshly calculated from new data: `df` and `indexConfig`. It doesn't read `logEntry` from older operation logs.

So there are only 2 cases:
1. logEntry creation fails because of incorrect config
2. logEntry creation passes because of correct config

And since logEntry is `lazy val`, it won't need any more creation in future `def event()` calls (which is where I meant it exists)

Reply:
@imback82